sahi/predict.py

```diff
@@ -579,9 +687,30 @@ def predict(
                 file_name=filename_without_extension,
                 export_format=visual_export_format,
             )
+
+            # Get the image with bboxes added on it with cv.py
+            image_and_elapsed_time = visualize_object_predictions(
+                np.ascontiguousarray(image_as_pil),
+                object_prediction_list=object_prediction_list,
+                rect_th=visual_bbox_thickness,
+                text_size=visual_text_size,
+                text_th=visual_text_thickness,
+                output_dir=output_dir,
+                file_name=filename_without_extension,
+                export_format=visual_export_format,
+            )
+            get_image_from_dict = image_and_elapsed_time["image"]
+
         time_end = time.time() - time_start
         durations_in_seconds["export_files"] = time_end
 
+        if export_video:
+            out.write(get_image_from_dict)
+        else:
+            Path(output_dir).mkdir(parents=True, exist_ok=True)
```

Comment:
`output_dir` is not defined when `export_visual=False`. Thus, this line fails when `export_visual=False`.

Reply:
I didn't try with `export_visual=False`. I will review this situation.